grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/RedisInflightTaskConsumer.java

```diff
@@ -92,26 +110,43 @@ public void run() {
     }
 
     private void processElement(String key, Jedis resource, String head) {
+        processed.mark();
         // TODO Use Jackson for this
         long runAt = Json.read(head).at("args").at(0).at("taskState")
                 .at("schedule").at("runAt").asLong();
         Instant runAtDate = Instant.ofEpochMilli(runAt);
         Duration gap = Duration.between(runAtDate, Instant.now());
+        if (gap.getSeconds() > destroyInterval.getSeconds()) {
+            destroyable.mark();
+            LOG.info("Found really old dead task in inflight, destroying it: {}", head);
+            attemptDestroy(key, resource);
+        }
         if (gap.getSeconds() > processInterval.getSeconds()) {
-            LOG.info("Found dead task in inflight: {}", head);
-            String keyDest = JesqueUtils
-                    .createKey(config.getNamespace(), QUEUE, queueName);
+            dead.mark();
+            LOG.info("Found dead task in inflight, moving it: {}", head);
```

Comment:
So what is a "dead" task anyway? How do we end up making them? 

Shouldn't we rather not make "dead" tasks rather then working around them?